crates/cxx-qt-gen/src/generator/rust/qobject.rs

```diff
@@ -18,9 +45,140 @@ pub struct GeneratedRustQObject {
     pub blocks: GeneratedRustQObjectBlocks,
 }
 
-pub struct GeneratedRustQObjectBlocks {
-    /// Module for the CXX bridge
-    pub cxx_mod_contents: Vec<Item>,
-    /// Items for the CXX-Qt module
-    pub cxx_qt_mod_contents: Vec<Item>,
+impl GeneratedRustQObject {
+    pub fn from(qobject: &ParsedQObject) -> Result<GeneratedRustQObject> {
+        // Create the base object
+        let qobject_idents = QObjectName::from(qobject);
+        let namespace_idents = NamespaceName::from(qobject);
+        let mut generated = GeneratedRustQObject {
+            cpp_struct_ident: qobject_idents.cpp_class.rust.clone(),
+            cxx_qt_thread_ident: qobject_idents.cxx_qt_thread_class.clone(),
+            namespace_internals: namespace_idents.internal,
+            rust_struct_ident: qobject_idents.rust_struct.rust.clone(),
+            blocks: GeneratedRustQObjectBlocks {
+                cxx_mod_contents: vec![],
+                cxx_qt_mod_contents: qobject.others.clone(),
+            },
+        };
+
+        // Add the QObject class definitions
+        generated.blocks.append(&mut generate(&qobject_idents)?);
+
+        // Add our QObject struct to the implementation blocks
+        generated
+            .blocks
+            .cxx_qt_mod_contents
+            .push(syn::Item::Struct(qobject.qobject_struct.clone().unwrap()));
+
+        // Generate methods for the properties, invokables, signals
+        generated.blocks.append(&mut generate_rust_properties(
+            &qobject.properties,
+            &qobject_idents,
+        )?);
+        generated.blocks.append(&mut generate_rust_invokables(
+            &qobject.invokables,
+            &qobject_idents,
+        )?);
+        generated.blocks.cxx_qt_mod_contents.extend_from_slice(
+            &qobject
+                .methods
+                .iter()
+                // Build non invokables on the C++ struct
+                .map(|method| {
+                    let cpp_class_name_rust = &qobject_idents.cpp_class.rust;
+                    syn::parse2(quote! {
+                        impl #cpp_class_name_rust {
+                            #method
+                        }
+                    })
+                })
+                .collect::<Result<Vec<Item>>>()?,
+        );
```

Comment:
This is the only code generation directly within this function. The rest of the function calls smaller functions that generate a specific part of the code. How about factoring this out to its own function?

Reply:
Done

Reply:
Do we want this in the qobject.rs file? or in the invokables.rs file? or in it's own methods.rs file?

Reply:
A separate file for a single, small function seems a bit overkill, but sure if you want :shrug:

Reply:
I think lets just leave it as a separate small function for now, if this file starts becoming larger we can split out.